compose/service.py

```diff
@@ -519,6 +522,59 @@ def pull(self, insecure_registry=False):
             )
 
 
+def get_container_data_volumes(container, volumes_option):
+    """Find the container data volumes that are in `volumes_option`, and return
+    a mapping of volume bindings for those volumes.
+    """
+    volumes = []
+
+    volumes_option = volumes_option or []
+    container_volumes = container.get('Volumes') or {}
+    original_volumes = set(volumes_option + container_volumes.keys())
+
+    for volume in original_volumes:
```

Comment:
This works in docker 1.2 to fix the test added in #859

Using docker 1.4.1 it seems like docker is refusing to override the volume with a `bind`.  This function is returning the correct value, but when `container.start(binds=...)` is called, the old volume remains.

Maybe `volumes_from` has to be used to override the volume? Maybe this is a bug in docker?  I'm not really clear on what the expected behaviour is.